### ISSUE TITLE
Automations are no longer exposed in Homekit

### ISSUE THREAD

**Issue by maisun** [ebaauw/homebridge-hue#1028]
Hi,
Thanks for the great plug-in, I use it to expose Hue automations (wake-up and go to sleep) as two switches in HomeKit. Recently I found those switches disappear in HomeKit, I’m not sure if it was due to the upgrade of the plug-in or maybe Hue app caused the issue. I have not touched the Homebridge configuration for quite some time. Any suggestion where to look? Thanks

**Comment by ebaauw** [comment]
Not sure what you mean by “automations”?  Hue bridge rules, or CLIPGenericFlag sensors used for meethue routines?  Double-check in the debug dump file, or in the Hue app, that these are still on the Hue bridge.  Then double-check in the debug logfile that Homebridge Hue is exposing them.

**Comment by maisun** [comment]
> Not sure what you mean by “automations”? Hue bridge rules, or CLIPGenericFlag sensors used for meethue routines? Double-check in the debug dump file, or in the Hue app, that these are still on the Hue bridge. Then double-check in the debug logfile that Homebridge Hue is exposing them.

Hi Erik, thank you for your reply! Not sure how to check if Hue Bridge still has them, I do see them in Hue app though, but after the new version I think they merged the "routines" with "Hue Labs". According to the log looks to me that the plugin didn't find anything not even a light:
`[9/22/2021, 9:33:33 PM] [homebridge-hue] Loaded homebridge-hue v0.13.23 child bridge successfully
[9/22/2021, 9:33:33 PM] Loaded 0 cached accessories from cachedAccessories.0EE9FEF5B4B4.
[9/22/2021, 9:33:33 PM] [homebridge-hue] homebridge-hue v0.13.23, node v14.17.1, homebridge v1.3.4
[9/22/2021, 9:33:33 PM] [homebridge-hue] warning: recommended version: node v14.17.6 LTS
[9/22/2021, 9:33:33 PM] [homebridge-hue] config.json: {"platform":"Hue","hosts":["192.***.***.36"],"users":{"001788FFFE716C42":"***************************************1"},"sensors":true,"nativeHomeKitSensors":false,"excludeSensorTypes":["Daylight","Geofence","CLIPGenericStatus","ZLLSwitch"],"linkButton":true}
[9/22/2021, 9:33:33 PM] [homebridge-hue] config: {"anyOn":true,"brightnessAdjustment":1,"effects":true,"excludeSensorTypes":{"Daylight":true,"Geofence":true,"CLIPGenericStatus":true,"ZLLSwitch":true,"ZHASwitch":true},"forceHttp":false,"heartrate":5,"hosts":["192.***.***.36"],"lowBattery":25,"nativeHomeKitLights":true,"nativeHomeKitSensors":false,"resetTimeout":500,"resource":true,"timeout":5,"users":{"001788FFFE716C42":"***************************************1"},"waitTimePut":50,"waitTimePutGroup":1000,"waitTimeResend":300,"waitTimeUpdate":20,"platform":"Hue","sensors":true,"linkButton":true}
[9/22/2021, 9:33:33 PM] [homebridge-hue] : request 1: GET /config
[9/22/2021, 9:33:33 PM] [homebridge-hue] : request 1: 200 OK
[9/22/2021, 9:33:33 PM] [homebridge-hue] Philips hue: Signify Netherlands B.V. BSB002 bridge v1947054040, api v1.46.0
[9/22/2021, 9:33:33 PM] [homebridge-hue] Philips hue: request 1: GET /
[9/22/2021, 9:33:33 PM] [homebridge-hue] Philips hue: request 1: 200 OK
[9/22/2021, 9:33:33 PM] [homebridge-hue] Philips hue: request 2: GET /groups/0
[9/22/2021, 9:33:33 PM] [homebridge-hue] Philips hue: request 2: 200 OK
[9/22/2021, 9:33:33 PM] [homebridge-hue] Philips hue: 001788FFFE716C42: Signify Netherlands B.V. BSB002 bridge "Philips hue"
[9/22/2021, 9:33:33 PM] [homebridge-hue] Philips hue: /sensors/1: Daylight excluded
[9/22/2021, 9:33:33 PM] [homebridge-hue] Philips hue: /sensors/4: ZLLSwitch excluded
[9/22/2021, 9:33:33 PM] [homebridge-hue] Philips hue: /sensors/7: ZLLSwitch excluded
[9/22/2021, 9:33:33 PM] [homebridge-hue] Philips hue: /sensors/98: CLIPGenericStatus excluded
[9/22/2021, 9:33:33 PM] [homebridge-hue] Philips hue: /sensors/102: CLIPGenericStatus excluded
[9/22/2021, 9:33:33 PM] [homebridge-hue] Philips hue: /sensors/103: CLIPGenericStatus excluded
[9/22/2021, 9:33:33 PM] [homebridge-hue] Philips hue: /sensors/104: CLIPGenericStatus excluded
[9/22/2021, 9:33:33 PM] [homebridge-hue] Philips hue: /sensors/105: CLIPGenericStatus excluded
[9/22/2021, 9:33:33 PM] [homebridge-hue] Philips hue: 0 sensors
[9/22/2021, 9:33:33 PM] [homebridge-hue] Philips hue: 0 lights
[9/22/2021, 9:33:33 PM] [homebridge-hue] Philips hue: 0 schedules
[9/22/2021, 9:33:33 PM] [homebridge-hue] Philips hue: 0 rules
[9/22/2021, 9:33:33 PM] [homebridge-hue] Philips hue: 0 groups
[9/22/2021, 9:33:33 PM] [homebridge-hue] Philips hue: 1 accessories
[9/22/2021, 9:33:33 PM] [homebridge-hue] masked debug info dumped to /homebridge/homebridge-hue.json.gz
[9/22/2021, 9:33:33 PM] Initializing platform accessory 'Philips hue'...
[9/22/2021, 9:33:33 PM] [homebridge-hue] Philips hue: 3 services
Preparing Advertiser for 'homebridge-hue 3DFB' using ciao backend!
[9/22/2021, 9:33:33 PM] [homebridge-hue] npm registry: request 1: GET /homebridge-hue/latest
Starting to advertise 'homebridge-hue 3DFB' using ciao backend!
[9/22/2021, 9:33:33 PM] Homebridge v1.3.4 (homebridge-hue) is running on port 48776.
[9/22/2021, 9:33:34 PM] [homebridge-hue] npm registry: request 1: 200 OK
[9/22/2021, 9:33:34 PM] [homebridge-hue] latest version: homebridge-hue v0.13.23`

**Comment by ebaauw** [comment]
They might have implemented the routines differently with the latest version, using CLIPGenericStatus sensors instead of CLIPGenericFlag?  I you attach the dump file, I could have a look.

**Comment by maisun** [comment]
> They might have implemented the routines differently with the latest version, using CLIPGenericStatus sensors instead of CLIPGenericFlag? I you attach the dump file, I could have a look.

Sorry but where can I find the dump file?

**Comment by ebaauw** [comment]
Yeah, that's it.  See also the FAQ.

I see five CLIPGenericStatus sensors: three `HUELABSVTOGGLE` (v2.0) and two `HUELABSENUM` (v1.0).  Each go to sleep formula uses a toggle and an enum; the colorloop formula only uses a toggle.  Looking at the rules, I suspect the toggle only takes values 0 and 1, but I don't understand why they don't / no longer use a CLIPGenericFlag for this.

If the toggle indeed only uses two values, I could expose it as a switch (just like a CLIPGenericFlag).

**Comment by maisun** [comment]
> Yeah, that's it. See also the FAQ.
> 
> I see five CLIPGenericStatus sensors: three `HUELABSVTOGGLE` (v2.0) and two `HUELABSENUM` (v1.0). Each go to sleep formula uses a toggle and an enum; the colorloop formula only uses a toggle. Looking at the rules, I suspect the toggle only takes values 0 and 1, but I don't understand why they don't / no longer use a CLIPGenericFlag for this.
> 
> If the toggle indeed only uses two values, I could expose it as a switch (just like a CLIPGenericFlag).

Ok looks like they've made some changes, it would be great to have switches in HomeKit to trigger those formulas. Just let me know if you want me to test the change, thank you very much in advance!

**Comment by ebaauw** [comment]
Could you try beta v0.13.24-0?  It should expose the toggles as switches.  Make sure to expose CLIPGenericStatus sensors, or whitelist the toggles (see https://github.com/ebaauw/homebridge-hue/wiki/Resource-Links).

**Comment by maisun** [comment]
> Could you try beta v0.13.24-0? It should expose the toggles as switches. Make sure to expose CLIPGenericStatus sensors, or whitelist the toggles (see https://github.com/ebaauw/homebridge-hue/wiki/Resource-Links).

Hi Erik,

Thanks a lot for quick fix! I can confirm the start/stop for the Hub Lab formulas work for me. HUELABSENUM is exposed as "dimming" devices and they don't seem to work, so I guess I need to find a way to hide them. I don't seem to get the wake-up formula to work though, it's exposed an "alarm", I recall previously it's exposed as a switch so I can automate wake-up routine in HomeKit. Please kindly find the screenshots and dump file (with wake-up formula) attached.
<img width="1067" alt="Screenshot 2021-09-25 at 09 09 48" src="https://user-images.githubusercontent.com/5745785/134763222-b3db3b0a-961e-49d4-8675-a9a8ed3322e4.png">
[homebridge-hue.json.gz](https://github.com/ebaauw/homebridge-hue/files/7229359/homebridge-hue.json.gz)

Have a great weekend!

**Comment by ebaauw** [comment]
> HUELABSENUM is exposed as "dimming" devices and they don't seem to work, so I guess I need to find a way to hide them. 

Use a `blacklist` resource line, see the Wiki.  They work just fine; Apple's home app just doesn't support them (see below).  Typically these CLIPGenericStatus sensors are used to record a state (0, 1, 2, ...) to keep track which step in a routine should be taken next.  The state is used to trigger a next rule, likely delayed, which then updates the state.

> I don't seem to get the wake-up formula to work though, it's exposed an "alarm", I recall previously it's exposed as a switch so I can automate wake-up routine in HomeKit.

The wake-up routine is triggered by a schedule, `/schedules/2`.  Set `schedules` in config.json to expose these to HomeKit, so you can en/disable the schedule from HomeKit.  Note that I don't expose schedules as switches, or Siri would act on them when you tell her to turn off a room.  They're exposed as separate services on the bridge accessory.

Schedules and CLIPGenericStatus sensors are exposed using custom HomeKit services and characteristics.  Apple's Home app doesn't support these; you need to use another app, like Eve.  If you create the HomeKit automations in Eve, they'll execute just fine.  If you create a HomeKit scene with custom characteristics in Eve, you can recall it from Home, or even from Siri.

**Comment by maisun** [comment]
Hi Erik,
Thank you for your reply!
> They work just fine; Apple's home app just doesn't support them (see below).
I also use Eve and I can see those dimming devices, they don't seem to response to my actions, but the status changes between 0 and 1. I haven't figure out a logic when they change though, it's not in sync with start/stop switches for sure. I'm reading the page on resource link but don't fully understand how does it work and how to black list those HUELABSENUM sensors. Do I have to run the command from the hosting machine of HomeBridge? And does it have any side effect if I black list the resources? Can I suggest in homebridge configuration allow users to differentiate between different types of CLIPGenericStatus (HUELABSENUM and HUELABSVTOGGLE) - it's much easier to exclude them there.

> Set schedules in config.json to expose these to HomeKit, so you can en/disable the schedule from HomeKit.
I enabled "schedules" and now I can turn on/off the schedule from Eve. I think it would be nice to find a way to manually trigger the wake-up routine, so I don't rely on the schedules in Hue app. The issue with schedules is that it needs to be turn on and off after the execution. I recall it was possible to have the switch to trigger the formula previously, not sure why Philips made such change. Do you know a better way of triggering turn on with brightness transition for 30mins (0% -> 75%)?

Appreciated your help on this!

**Comment by ebaauw** [comment]
> Do you know a better way of triggering turn on with brightness transition for 30mins (0% -> 75%)?

The way I've implemented this: use a CLIPGenericFlag as virtual switch to trigger the wakeup.  I've created three rules:
- One that triggers immediately when the flag is set, and turns on the lights at minimum brightness.  Next to that, I run a HomeKit automation to turn on the bedroom speakers;
- One that triggers 1 second after the flag has been set, and turns on the lights to maximum brightness, with a transition time of 10 minutesl
- One that triggers 10:01 minutes after the flag has been set, and turns on regular room automation: opening the curtains when the sun has already risen, and turn the lights off when it's light enough (after the curtains are open), or on when it gets too dark (in the evening, during a thunderstorm or solar eclipse - although I still haven't tested that last scenario).

I actually use two bridge schedules to set the flag: one for workdays and one for the weekend.  Occasionally I set the flag from a HomeKit schedule, if I need a different wakeup time.  In this case I (temporary) disable the bridge schedule.
